### ISSUE TITLE
pentadactyl hints don't work on new warehouse web interface

### ISSUE THREAD

**Issue by bitdancer** [pypa/warehouse#785]
Pentadactyl (http://5digits.org/pentadactyl/) is a firefox plugin that allows firefox to be used almost exclisively from the keyboard.  One of the core features is hints: you press 'f' and every link on the page gets highlighted with a number in front of it, and you can type the number to select the link.  This works with most but not all websites, and I've never bothered to figure out why some don't work.  The new warehouse is one that doesn't, and unlike most web sites where I run into this, *every* link does not get a label (usually it is just some that don't).  I'm sure I'm a minority audience so I won't be upset if you decide not to deal with this issue :)

**Comment by dstufft** [comment]
My biggest concern is that whatever is causing Pentadactyl to fail would also cause someone using a screen reader or something to fail (and thus it'd be an accessibility problem as well).

**Comment by bitdancer** [comment]
That's a good point, and I wouldn't be surprised if it was the case.  Also, I believe there is at least one KDE (or is it gnome?) browser that has a similar feature.  It would be interesting to know if that failed, also.

**Comment by sigmavirus24** [comment]
So the screen hints don't display (but using them you can tab between links so it's not an HTML problem). It's likely because of the CSS choice which is unrelated to what a screen reader would use. Is there heavy use of `!important` in the CSS here? If so, I can imagine that causing problems with the CSS used to overlay the hints visually.

**Comment by dstufft** [comment]
Ah, in that case it's a design problem. I'll let Nicole take a poke and see how hard it looks to fix it and if she wants to support it or not.

**Comment by sigmavirus24** [comment]
(Also this was present pre-redesign too)

**Comment by brainwane** [comment]
@bitdancer Thanks for your bug report, and sorry for the slow response!

For context: the folks working on Warehouse have gotten [funding to concentrate on improving and deploying it](https://pyfound.blogspot.com/2017/11/the-psf-awarded-moss-grant-pypi.html), and have kicked off work towards [our development roadmap](https://wiki.python.org/psf/WarehouseRoadmap) -- the most urgent task is to improve Warehouse to the point where we can redirect pypi.python.org to pypi.org so the site is more sustainable and reliable. And we're taking some time to take a look at older issues in the repo as we go.

If you have a chance to check out the current pre-production site, http://pypi.org/ , with your preferred keyboard-first setup, and let us know whether you still run into these kinds of problems, I'd appreciate it.

I've put this issue into [the "shut down legacy PyPI" milestone](https://github.com/pypa/warehouse/milestone/7) because I want to make sure we address it before we shut down the old site.

Thanks and sorry again for the wait.

**Comment by sigmavirus24** [comment]
For what it's worth, unless you're on the really old ESR of Firefox, Pentadactyl is no longer maintained for new versions of Firefox. I wouldn't think this is necessary at all at this point. Eventually the ESR will become unsupported and thus pentadactyl will finally be killed off. Further, it was likely already a very small margin of users.

**Comment by bitdancer** [comment]
Yeah, I'm currently using the ESR, pretty much until I'm forced to
find something to replace pentadactyl.   But I can understand you
not wanting to support something that is going to die, so I'm fine
if you close this.

**Comment by brainwane** [comment]
@bitdancer If you could do some checking and see what alternatives to pentadactyl you're likely to switch to, I would appreciate it -- I'd like for us to test with those.

Also, just so you know, we're also seeking a grant that would, among other things, support more accessibility testing and improvements to Warehouse.

**Comment by brainwane** [comment]
I [just spoke with Nicole in IRC](http://kafka.dcpython.org/day/pypa-dev/2018-03-21) and we agreed that, given how much better Warehouse is on accessibility in general compared to legacy PyPI, this improvement can wait till after we shut down legacy PyPI.
